### PR TITLE
cool#10630 doc electronic sign: send the hash on the server

### PR DESCRIPTION
**Pull request by vmiklos**
- **cool#10630 doc electronic sign: expose the name of the document**
- **wsd: fix clang 18 -Werror,-Wvla-cxx-extension**
- **cool#10630 doc electronic sign: send the hash on the server**

### CODE REVIEW

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_android_app_co-24.04/2810/console fails because I forgot that it seems our mobile apps don't do any networking, so just put that inside a mobile ifdef for now. Perhaps as a next step it'll be useful to add a wsd/ClientSession-$PLATFORM.cpp similar to e.g. common/Util-mobile.cpp, to avoid the ifdefs.

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04/3661/consoleFull failed in unit-storage and unit-storage but both pass for me locally, retry.

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04/3662/consoleFull hmm, same failure here, retry.

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04_cypress_desktop/3112/console had 5 failures: impress/undo_redo_spec.js, writer/annotation_spec.js, writer/navigator_spec.js, writer/scrolling_spec.js and writer/searchbar_spec.js, retry to see if this is persistent.

**Comment by vmiklos** [comment]
Ah, I could not reproduce the failure locally because my core was too old: https://gerrit.libreoffice.org/c/core/+/177957 is meant to help here.

**Comment by vmiklos** [comment]
After rebase, https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04/3691/console fails due to a timeout, hmm.

**Comment by vmiklos** [comment]
@caolanm could you please review this? Thanks.

The point is that the credentials are a pair of client ID + secret, so secret is needed to initiate a sign (which has costs; results in a "document ID"), but the browser can do the signing without the secret now. This will also allow undoing the CSP rules for that specific domain, which I always disliked.

The CI is still in a bad shape, but this one at least passed now.

File: wsd/ClientSession.cpp [hunk 1]
```diff
@@ -428,6 +428,90 @@ void ClientSession::onTileProcessed(TileWireId wireId)
...
+
+    // Make the request: this requires a JSON body, where we set the secret
+    std::string commandArguments = tokens.cat(' ', 2);
+    Poco::JSON::Object::Ptr commandArgumentsObject;
+    if (!JsonUtil::parseJSON(commandArguments, commandArgumentsObject))
+    {
+        LOG_WRN("PrepareSignature: failed to parse arguments as JSON");
+        return false;
+    }
+    auto requestBodyObject = commandArgumentsObject->get("body").extract<Poco::JSON::Object::Ptr>();
```
Comment (caolanm) [inline wsd/ClientSession.cpp:498]: FWIW I think (didn't verify though) that get + extract will throw if "body" doesn't exist, while getObject("body") will return null to allow an explicit test for its existence. Not sure if that unlikely case matters here, and if it does then maybe that throw is perfectly acceptable.

**Review by caolanm — APPROVED** [review]

File: wsd/ClientSession.cpp (hunk 1 above)
Comment (vmiklos) [reply wsd/ClientSession.cpp:498]: This is something we control, so I would say it's OK if this throws in the unlikely case that our JS code at browser/src/control/Control.ESignature.ts:230 produces an unexpected JSON.

But you have a point here, I'll see if I can rework this to use JsonUtil::findJSONValue() instead, which won't throw, even if the key is missing. Thanks.

**Pull request merged**